### PR TITLE
repo: adopt `rust-lang` label system

### PR DESCRIPTION
**Pull request by hinto-janai**
Proposal to replace the current set of labels with the label system used by `rust-lang` and `tokio`.

Inspiration:
- https://rustc-dev-guide.rust-lang.org/contributing.html#issue-triage
- https://github.com/tokio-rs/tokio/blob/master/CONTRIBUTING.md#keeping-track-of-issues-and-prs

This PR adds a guide to `CONTRIBUTING.md` - these labels will be created if/when this PR is merged. This also adds [`labeler.yml`](https://github.com/actions/labeler) which automatically labels things based off changed file paths.

List of labels:

| Label          | Description |
|----------------|-------------|
| A-consensus    | consensus/
| A-cryptonight  | cryptonight/
| A-database     | database/
| A-helper       | helper/
| A-net          | net/
| A-p2p          | p2p/
| A-pruning      | pruning/
| A-test-utils   | test-utils/
| A-types        | types/
| A-workspace    | Changes to any repo root file.
| A-ci           | Changes to `.github`
| A-dependency   | Changes to any `Cargo.{toml,lock}` file
| A-docs         | Changes to any `.md` file
| C-bug          | This is a bug-report. Bug-fix PRs use C-fix instead.
| C-fix          | PRs that fixes code, or issues documenting a fix.
| C-cleanup      | PRs that clean code up or issues documenting cleanup.
| C-discussion   | Discussion or questions that doesn't represent real issues.
| C-optimization | An issue highlighting optimization opportunities or PRs implementing such.
| C-enhancement  | This is a PR that adds a new features
| C-proposal     | A proposal of some kind, and a request for comments.
| C-question     | A user question. Large overlap with GitHub discussions.
| C-request      | A request for changes or a feature.
| C-regression   | Fixes/tracks code regressions.
| D-incorrect    | A diagnostic that is giving misleading or incorrect information.
| D-terse        | An code/error/message that doesn't give enough information about the problem at hand.
| D-confusing    | Confusing code/error/message that should be reworked.
| D-inconsistent | Inconsistency in formatting, grammar or style between diagnostic messages.
| D-verbose      | Too much output caused by a single piece of incorrect code.
| E-easy         | Easy difficulty. Experience needed to fix: Not much. Good first issue.
| E-medium       | Medium difficulty. Experience needed to fix: Intermediate.
| E-hard         | Hard difficulty. Experience needed to fix: A lot.
| I-crash        | Cuprate crashes (SIGSEGV, SIGABRT, etc). Use I-ice instead for panics.
| I-panic        | Cuprate panics. Use I-crash instead for crashes (SIGSEGV, SIGABRT, etc).
| I-unsound      | A soundness hole, see: https://en.wikipedia.org/wiki/Soundness
| I-hang         | Cuprate never terminates, due to infinite loops, deadlock, livelock, etc.
| I-size         | Problems and improvements with respect to binary size of generated code.
| I-perf         | Problems and improvements with respect to performance of code.
| I-memory       | Large unexpected memory usage and/or memory leaks
| O-windows      | Windows specific issue
| O-macos        | macOS specific issue
| O-linux        | Linux specific issue
| P-low          | Low priority
| P-medium       | Medium priority
| P-high         | High priority
| P-critical     | Critical priority
| wontfix        | Indicates an issue will not be fixed.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- database/*
+
+A-helper:
+- helper/*
+
+A-net:
+- net/*
+
+A-net:
+- net/*
```
Comment (Boog900) [inline .github/labeler.yml:17]: Is this supposed to be duplicated?

**Review by Boog900 — CHANGES_REQUESTED** [review]
To add labels do you just add them to `labeler.yml`, whats the process?

File: .github/labeler.yml (hunk 1 above)
Comment (hinto-janai) [reply .github/labeler.yml]: Nope, fixed

**Comment by hinto-janai** [comment]
> To add labels do you just add them to labeler.yml, whats the process?

I added some docs to `.github/labeler.yml`.

Basically you define patterns that when [labeler](https://github.com/actions/labeler) matches in CI, it will automatically add/remove labels from a PR.

Examples here: https://github.com/hinto-janai/labeler-test/pulls

**Review by Boog900 — APPROVED** [review]
Looks good!

**Pull request merged**